src/wasm/wasm-binary.cpp

```diff
@@ -2137,7 +2137,17 @@ void WasmBinaryBuilder::readFunctions() {
       assert(controlFlowStack.empty());
       assert(letStack.empty());
       assert(depth == 0);
-      func->body = getBlockOrSingleton(func->sig.results);
+      if (!skipFunctionBodies) {
+        func->body = getBlockOrSingleton(func->sig.results);
+      } else {
+        // When skipping the function body we need to put something valid in
+        // their place so we validate. An unreachable is always acceptable
+        // there.
+        func->body = Builder(wasm).makeUnreachable();
```

Comment:
Can we mark the function in such as way that if anyone tries to walk it we will assert?

Reply:
There isn't a mechanism for that - we'd need to add a new property on functions. Or a new instruction "Unwalkable" perhaps? I'm not sure it's worth either though, as an error here would not be a subtle bug, it would break very loudly.

Reply:
I think it might break silently.   What if I write a new pass in `wasm-emscripten-finalize` that searches for a certain instruction.   With this change that pass will silently always fail, right?

Reply:
It will silently always return "false" I suppose, but that's what I meant by "loudly". But I guess there could be a subtle new pass in which it's not as loud as it would currently be, fair point...